utils/remote_control/navigator_emergency_control/nodes/nodes/navigator_emergency.py

```diff
@@ -0,0 +1,137 @@
+
+            # No change in state
+            if rospy.Time.now() - self.last_joy.header.stamp > rospy.Duration(15 * 60):
+
+                # The controller times out after 15 minutes
+                if self.active:
+                    rospy.logwarn("Controller Timed out. Hold start to resume.")
+                    self.reset()
+
+        else:
+            joy.header.stamp = rospy.Time.now()  # In the sim, stamps weren't working right
+            self.last_joy = joy
+
+    def joy_recieved(self, joy):
+        self.check_for_timeout(joy)
+
+        # Assigns readable names to the buttons that are used
+        start = joy.buttons[7]
+        kill = bool(joy.buttons[2])
+        station_hold = bool(joy.buttons[0])
+        emergency_control = bool(joy.buttons[11])
+        auto_control = bool(joy.buttons[12])
+        shooter_cancel = bool(joy.buttons[1])
+        change_mode = bool(joy.buttons[3])
+
+        # Reset controller state if only start is pressed down about 3 seconds
+        self.start_count += start
+        if self.start_count > 10:
+            rospy.loginfo("Resetting controller state")
+            self.reset()
+            self.active = True
+            self.remote.clear_kill()
+
+        if not self.active:
+            return
+
+        if kill and not self.last_kill:
+            self.remote.toggle_kill()
+
+        if station_hold and not self.last_station_hold_state:
+            self.remote.station_hold()
+
+        if auto_control and not self.last_auto_control:
+            self.remote.select_autonomous_control()
+
+        if emergency_control and not self.last_emergency_control:
+            self.remote.select_emergency_control()
+
+        if shooter_cancel and not self.last_shooter_cancel:
+            self.remote.shooter_cancel()
+
+        if change_mode and not self.last_change_mode:
+            self.remote.select_next_control()
+
+        self.last_kill = kill
+        self.last_station_hold_state = station_hold
+        self.last_auto_control = auto_control
+        self.last_emergency_control = emergency_control
+        self.last_shooter_cancel = shooter_cancel
+        self.last_change_mode = change_mode
+
+        # Scale joystick input to force and publish a wrench
+        x = joy.axes[1] * self.force_scale
+        y = joy.axes[0] * self.force_scale
+        rotation = joy.axes[3] * self.torque_scale
+        self.remote.publish_wrench(x, y, rotation, joy.header.stamp)
+
+    def timeout(self, event):
+        '''
+        Publishes zeros after 2 seconds of no update 
+	in case node navigator_emergency_xbee dies
+        '''
+        if self.active:
+
+	    # No change in state
+	    if rospy.Time.now() - self.last_joy.header.stamp > rospy.Duration(2):
+
+	        # No new instructions after 2 seconds, zero the wrench
+                self.reset()
+
+if __name__ == "__main__":
+    emergency = Joystick()
+    rospy.Timer(rospy.Duration(1), emergency.timeout, oneshot=False)
```

Comment:
Yes, but the duration within the function is set to 2s. If the timer were, say, 1ms, it would check every ms if the time interval between now and the last received joy is greater than 2s. As it is right now, it does that every second, so I suppose it's really activating between 2-3s. Setting the timer to 2s would be anywhere between 2-4s. I could make it more accurate by making the timer a smaller duration, but I don't think there's any need to run it very often for the sake of accuracy.